docs/contributor/adr/0009-egress-gateway.md

```diff
@@ -0,0 +1,97 @@
+# Support for **egressGateway** Configuration
+
+## Status
+
+Accepted
+
+## Context
+
+To support cluster configurations that require control of outbound traffic
+besides routing of incoming traffic, Istio module needs to support `egressGateway`.
+This Istio component allows for intercepting traffic coming from in-mesh pods to 
+targets outside of the cluster scope. This allows users to perform tasks that include
+monitoring and securing access to outbound resources.
+
+## Decision
+
+We will introduce configuration for `egressGateway` Istio component in the
+Istio CustomResource (CR). The API will allow to conditionally enable
+the `egressGateway` as well as configuration for [KubernetesResourcesSpec(k8s)](https://istio.io/latest/docs/reference/config/istio.operator.v1alpha1/#KubernetesResourcesSpec).
+
+## Consequences
+
+The Istio CustomResource API will be extended by configuration related to egressGateway. The `components` section will now include additional `egressGateway` field, that will contain a boolean `enabled` flag, as well as configuration for `KubernetesResourcesSpec(k8s)`. The `k8s` configuration will have the exact same structure as for `ingressGateway`.
```

Comment:
Let's underline that it will be disabled by default, which is different than in the ingressGateway